### PR TITLE
boards controller infractruscture reopened

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

File: src/DashboardService.Mappers/Db/DbBoardMapper.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  public DbBoard Map(CreateBoardRequest request)
+  {
+    return new()
+    {
+      Id = Guid.NewGuid(),
+      ProjectId = request.ProjectId,
+      Name = request.Name,
+      IsActive = true,
+      CreatedBy = _httpContextAccessor.HttpContext.GetUserId(),
```
Comment (VLiamin) [inline src/DashboardService.Mappers/Db/DbBoardMapper.cs]: @SolovyovDmitry Do we pass the user Id to the mapper or do it here?

Comment (SolovyovDmitry) [reply src/DashboardService.Mappers/Db/DbBoardMapper.cs]: yep, pass

File: src/DashboardService.Business/Board/GetAllBoardsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using LT.DigitalOffice.DashboardService.Models.Dto.Models;
+using LT.DigitalOffice.Kernel.Responses;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.Board;
+
+public class GetAllBoardsCommand : IGetAllBoardsCommand
+{
+  public Task<FindResultResponse<BoardInfo>> ExecuteAsync()
```
Comment (SolovyovDmitry) [inline src/DashboardService.Business/Board/GetAllBoardsCommand.cs]: filter

File: src/DashboardService.Data/BoardRepository.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  {
+    _provider = provider;
+  }
+
+  public Task CreateAsync(DbBoard board)
+  {
+    throw new NotImplementedException();
+  }
+
+  public Task<bool> EditByIdAsync(Guid id)
```
Comment (SolovyovDmitry) [inline src/DashboardService.Data/BoardRepository.cs]: patch request

File: src/DashboardService.Models.Dto/Requests/Board/CreateBoardRequest.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace LT.DigitalOffice.DashboardService.Models.Dto.Requests.Board;
+
+public record class CreateBoardRequest
+{
+  public Guid ProjectId { get; set; }
```
Comment (SolovyovDmitry) [inline src/DashboardService.Models.Dto/Requests/Board/CreateBoardRequest.cs:9]: it should be either nullable or required

**Review by SolovyovDmitry — APPROVED** [review]

File: src/DashboardService.Mappers/Db/DbBoardMapper.cs (hunk 1 above)
Comment (YariLAN) [reply src/DashboardService.Mappers/Db/DbBoardMapper.cs]: +

File: src/DashboardService.Business/Board/GetAllBoardsCommand.cs (hunk 2 above)
Comment (YariLAN) [reply src/DashboardService.Business/Board/GetAllBoardsCommand.cs]: +

File: src/DashboardService.Data/BoardRepository.cs (hunk 3 above)
Comment (YariLAN) [reply src/DashboardService.Data/BoardRepository.cs]: +

File: src/DashboardService.Models.Dto/Requests/Board/CreateBoardRequest.cs (hunk 4 above)
Comment (YariLAN) [reply src/DashboardService.Models.Dto/Requests/Board/CreateBoardRequest.cs:9]: +

File: src/DashboardService.Business/Board/CreateBoardCommand.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  private readonly IAccessValidator _accessValidator;
+  private readonly IResponseCreator _responseCreator;
+
+  public CreateBoardCommand(
+    IBoardRepository repository,
+    IDbBoardMapper dbBoardMapper,
+    IHttpContextAccessor httpContext,
+    IAccessValidator accessValidator,
+    IResponseCreator responseCreator
+    )
```
Comment (VLiamin) [inline src/DashboardService.Business/Board/CreateBoardCommand.cs]: on previous

File: src/DashboardService.Business/Board/EditBoardCommand.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+namespace LT.DigitalOffice.DashboardService.Business.Board;
+
+public class EditBoardCommand : IEditBoardCommand
+{
+  private readonly IBoardRepository _boardRepository;
+  private readonly IPatchDbBoardMapper _patchDbBoardMapper;
+  private readonly IHttpContextAccessor _httpContext;
+  private readonly IAccessValidator _accessValidator;
+  private readonly IResponseCreator _responseCreator;
+
```
Comment (VLiamin) [inline src/DashboardService.Business/Board/EditBoardCommand.cs:22]: empty line

File: src/DashboardService.Business/Board/CreateBoardCommand.cs (hunk 5 above)
Comment (YariLAN) [reply src/DashboardService.Business/Board/CreateBoardCommand.cs]: +

File: src/DashboardService.Business/Board/EditBoardCommand.cs (hunk 6 above)
Comment (YariLAN) [reply src/DashboardService.Business/Board/EditBoardCommand.cs:22]: +

**Review by VLiamin — APPROVED** [review]

**Pull request merged**